aws/data_source_aws_route53_resolver_endpoint_test.go

```diff
@@ -0,0 +1,217 @@
+					resource.TestCheckResourceAttrPair(datasourceName, "id", resourceName, "id"),
+					resource.TestCheckResourceAttr(datasourceName, "ip_addresses.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsRoute53ResolverEndpoint_Filter(t *testing.T) {
+	name := acctest.RandomWithPrefix("tf-acc-test")
+	rInt := acctest.RandInt()
+	direction := "OUTBOUND"
+	resourceName := "aws_route53_resolver_endpoint.foo"
+	datasourceName := "data.aws_route53_resolver_endpoint.foo"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceAwsRoute53ResolverEndpointConfig_NonExistentFilter,
+				ExpectError: regexp.MustCompile("Your query returned no results. Please change your search criteria and try again"),
+			},
+			{
+				Config: testAccDataSourceRoute53ResolverEndpointConfig_filter(rInt, direction, name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(datasourceName, "name", resourceName, "name"),
+					resource.TestCheckResourceAttrPair(datasourceName, "id", resourceName, "id"),
+					resource.TestCheckResourceAttr(datasourceName, "ip_addresses.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceRoute53ResolverEndpointConfig_base(rInt int) string {
+	return fmt.Sprintf(`
+resource "aws_vpc" "foo" {
+  cidr_block           = "10.0.0.0/16"
+  enable_dns_support   = true
+  enable_dns_hostnames = true
+
+  tags = {
+    Name = "terraform-testacc-r53-resolver-vpc-%d"
+  }
+}
+
+data "aws_availability_zones" "available" {}
+
+resource "aws_subnet" "sn1" {
+  vpc_id            = aws_vpc.foo.id
+  cidr_block        = cidrsubnet(aws_vpc.foo.cidr_block, 2, 0)
+  availability_zone = data.aws_availability_zones.available.names[0]
+
+  tags = {
+    Name = "tf-acc-r53-resolver-sn1-%d"
+  }
+}
+
+resource "aws_subnet" "sn2" {
+  vpc_id            = aws_vpc.foo.id
+
+  cidr_block        = cidrsubnet(aws_vpc.foo.cidr_block, 2, 1)
+  availability_zone = data.aws_availability_zones.available.names[1]
+
+  tags = {
+    Name = "tf-acc-r53-resolver-sn2-%d"
+  }
+}
+
+resource "aws_subnet" "sn3" {
+  vpc_id            = aws_vpc.foo.id
+  cidr_block        = cidrsubnet(aws_vpc.foo.cidr_block, 2, 2)
+  availability_zone = data.aws_availability_zones.available.names[2]
+
+  tags = {
+    Name = "tf-acc-r53-resolver-sn3-%d"
+  }
+}
+
+resource "aws_security_group" "sg1" {
+  vpc_id = aws_vpc.foo.id
+  name   = "tf-acc-r53-resolver-sg1-%d"
+
+  tags = {
+    Name = "tf-acc-r53-resolver-sg1-%d"
+  }
+}
+
+resource "aws_security_group" "sg2" {
+  vpc_id = aws_vpc.foo.id
+  name   = "tf-acc-r53-resolver-sg2-%d"
+
+  tags = {
+    Name = "tf-acc-r53-resolver-sg2-%d"
+  }
+}
+`, rInt, rInt, rInt, rInt, rInt, rInt, rInt, rInt)
+}
+
+func testAccDataSourceRoute53ResolverEndpointConfig_initial(rInt int, direction, name string) string {
+	return fmt.Sprintf(`
+%s
```

Comment:
Can use [`composeConfig`](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#combining-test-configurations).